src/lib.rs

```diff
@@ -2610,3 +2621,62 @@ impl SwapChain {
         }
     }
 }
+
+type ErrorSinkCallback = dyn Fn(GPUError) ;
+type ErrorSink = Arc<Mutex<ErrorSinkRaw>>;
+
+
+/// TODO:
+#[derive(Debug)]
+pub struct GPUError {
+    source: Box<dyn Error + Send + Sync + 'static>
+}
+impl Display for GPUError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.source, f)
+    }
+}
+
+impl <E:Error+Send+Sync+'static> From<E> for GPUError {
+    fn from(err: E) -> Self {
+        GPUError {
+            source: err.into()
+        }
+    }
+}
+
+struct ErrorSinkRaw
+{
+
+    uncaught_handler: Box<ErrorSinkCallback>,
+
+    // error_scopes: Vec<Box<dyn std::future::Future<Output=Option<GPUError>>>>
+    // handler_stack: Vec<Box<ErrorSinkCallback>>
+}
+
+impl ErrorSinkRaw {
+    fn new() -> ErrorSinkRaw {
+        ErrorSinkRaw {
+            uncaught_handler: Box::from(default_error_handler),
+            
+            // handler_stack: Vec::new(),
+        }
+    }
+
+//    fn handle_error<E:Error+Send+Sync+'static>(&self, err: E) {
+    fn handle_error(&self, err: GPUError) {
+        // let handler = self.handler_stack.last().unwrap_or(&self.default_handler);
+        (self.uncaught_handler)(err);
+    }
+}
+
+impl Debug for ErrorSinkRaw {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f,"ErrorSink")
+    }
+}
+
+fn default_error_handler(err: GPUError) {
+    eprintln!("WGPU Error: {}",err);
```

Comment:
any reason this isn't a `panic` right away?

Reply:
For whatever reason, I couldn't get it to print anything but "Box<Any>" for the panic message. Though I kind of prefer it like this anyway, the actual error is somewhat easier to read, but I would be fine either way.

Reply:
I don't mind either way